grafana/main.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strings"
+	"text/template"
+)
+
+type Context struct {
+}
+
+func (c *Context) Env() map[string]string {
+	env := make(map[string]string)
+	for _, i := range os.Environ() {
+		sep := strings.Index(i, "=")
+		env[i[0:sep]] = i[sep+1:]
+	}
+	return env
+}
+
+func main() {
+	influxDbUrl := os.Getenv("INFLUXDB_URL")
+	if len(influxDbUrl) == 0 && len(os.Getenv("INFLUXDB_SERVICE_NAME")) > 0 {
+		os.Setenv("INFLUXDB_URL", "")
+		influxDbServiceUrl := "${INFLUXDB_PROTO}://${" +
+			os.Getenv("INFLUXDB_SERVICE_NAME") + "_SERVICE_HOST}:${" +
+			os.Getenv("INFLUXDB_SERVICE_NAME") + "_SERVICE_PORT}/"
```

Comment:
We can take in as input the influxdb service name and rely on DNS instead. That approach avoids requiring an influxdb service to running prior to deploying Grafana (This is very useful when run `kubectl.sh create -f configs/`)